chore(build): Add OpenRewrite plugin to root POM

**Is your feature request related to a problem? Please describe.**
When using Open Rewrite for cleanup actions I usually run the Maven CLI, recipe by recipe, module by module. Running the CLI on root level usually fails.

E.g. running:
```
mvn -U org.openrewrite.maven:rewrite-maven-plugin:run -Drewrite.activeRecipes=org.openrewrite.java.RemoveUnusedImports -Drewrite.exportDatatables=true
```

fails with
```
[INFO] --- compiler:3.13.0:compile (default-compile) @ operaton-engine-feel-juel ---
[INFO] Recompiling the module because of changed source code.
[INFO] Compiling 29 source files with javac [debug target 17] to target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/kthoms/Development/git/github/operaton/engine-dmn/feel-juel/src/main/java/org/operaton/bpm/dmn/feel/impl/juel/FeelEngineFactoryImpl.java:[23,45] package org.operaton.bpm.impl.juel.jakarta.el does not exist
[ERROR] /Users/kthoms/Development/git/github/operaton/engine-dmn/feel-juel/src/main/java/org/operaton/bpm/dmn/feel/impl/juel/FeelEngineFactoryImpl.java:[24,45] package org.operaton.bpm.impl.juel.jakarta.el does not exist
[ERROR] /Users/kthoms/Development/git/github/operaton/engine-dmn/feel-juel/src/main/java/org/operaton/bpm/dmn/feel/impl/juel/FeelEngineFactoryImpl.java:[91,13] cannot find symbol
````

This is due to steps that require phases beyond compile, here it is related to #84.

With fix for #84 it would fail with
```
[ERROR] Failed to execute goal org.eclipse.transformer:transformer-maven-plugin:1.0.0:transform (transform-java-sources) on project operaton-engine-spring-6: Exception transforming directory: Exception loading transformer rules: /Users/kthoms/Development/git/github/operaton/engine/target/classes (Is a directory) -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.transformer:transformer-maven-plugin:1.0.0:transform (transform-java-sources) on project operaton-engine-spring-6: Exception transforming directory
```

**Describe the solution you'd like**
Add a profile with the Open Rewrite plugin as [documented](https://docs.openrewrite.org/running-recipes/multi-module-maven).

Cleanups should then be executable by adding active recipes to the plugin config or to a `rewrite.yaml`, and then execute `./mvnw rewrite:run`.

Then incrementally add active recipes that improve the code quality. e.g. starting with ["Remove unused imports"](https://docs.openrewrite.org/recipes/java/removeunusedimports).

Whenever a new recipe is added make sure that it was run once against the code base and its changes have been applied.

**Describe alternatives you've considered**
It is of course still possible to run the Maven CLI if adding the profile is not feasible.

--- a/pom.xml
+++ b/pom.xml
@@ -529,6 +529,41 @@
         </plugins>
       </build>
     </profile>
+
+    <profile>
+      <id>openrewrite</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.openrewrite.maven</groupId>
+            <artifactId>rewrite-maven-plugin</artifactId>
+            <version>5.45.0</version>
+            <configuration>
+              <activeRecipes>
+                org.openrewrite.java.RemoveUnusedImports
+              </activeRecipes>
+              <failOnDryRunResults>true</failOnDryRunResults>
+              <exclusions>
+                **/node/**
+              </exclusions>
+            </configuration>
+            <dependencies>
+              <dependency>
+                <groupId>org.openrewrite.recipe</groupId>
+                <artifactId>rewrite-static-analysis</artifactId>
+                <version>1.20.0</version>
+              </dependency>
+              <dependency>
+                <groupId>org.openrewrite.recipe</groupId>
+                <artifactId>rewrite-testing-frameworks</artifactId>
+                <version>2.22.0</version>
+              </dependency>
+            </dependencies>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+
   </profiles>
 
   <repositories>

